test/gui/shared/scripts/bdd_hooks.py

```diff
@@ -93,6 +93,24 @@ def hook(context):
 
 @OnScenarioEnd
 def hook(context):
+    # capture screenshot if there is error in the scenario execution, and if the test is being run in CI
+    if test.resultCount("errors") > 0 and os.getenv('CI'):
+        import gi
```

Comment:
shouldn't imports be on the top of the file?

Reply:
Not necessarily. We can import the library only when we need it. We do not need it unless there are errors.